openeo_processes_dask/process_implementations/arrays.py

```diff
@@ -182,3 +182,104 @@ def array_labels(data: ArrayLike) -> ArrayLike:
     if len(data.shape) > 1:
         raise TooManyDimensions("array_labels is only implemented for 1D arrays.")
     return np.arange(len(data))
+
+
+def order(
+    data: ArrayLike,
+    asc: Optional[bool] = True,
+    nodata: Optional[bool] = None,
+    axis: Optional[int] = None,
+):
+    if isinstance(data, list):
+        data = np.asarray(data)
+    if len(data) == 0:
+        return data
+
+    # See https://github.com/dask/dask/issues/4368
+    logger.warning(
+        "order: Dask does not support lazy sorting of arrays, therefore the array is loaded into memory here. This might fail for arrays that don't fit into memory."
+    )
+
+    if asc:
+        permutation_idxs = np.argsort(data, kind="mergesort", axis=axis)
+    else:  # [::-1] not possible
+        permutation_idxs = np.argsort(
+            -data, kind="mergesort", axis=axis
+        )  # to get the indizes in descending order, the sign of the data is changed
+
+    if nodata is None:  # ignore np.nan values
+        if len(data.shape) > 1:
+            raise ValueError(
+                "order with nodata=None is not supported for arrays with more than one dimension, as this would result in sparse multi-dimensional arrays."
+            )
+        # sort the original data first, to get correct position of no data values
+        sorted_data = np.take_along_axis(data, permutation_idxs, axis=axis)
+        return permutation_idxs[~pd.isnull(sorted_data)]
+    elif nodata is False:  # put location/index of np.nan values first
+        # sort the original data first, to get correct position of no data values
+        sorted_data = data[permutation_idxs]
+        return np.append(
+            permutation_idxs[pd.isnull(sorted_data)],
+            permutation_idxs[~pd.isnull(sorted_data)],
+        )
+    elif nodata is True:  # default argsort behaviour, np.nan values are put last
+        return permutation_idxs
+
+
+def rearrange(
+    data: ArrayLike,
+    order: ArrayLike,
+    axis: Optional[int] = None,
+    source_transposed_axis: int = None,
+):
+    if len(data) == 0:
+        return data
+    if isinstance(data, list):
+        data = np.asarray(data)
+    if isinstance(order, list):
+        order = np.asarray(order)
+
+    if len(data.shape) != len(order.shape):
```

Comment:
Hmm - the spec seems ambiguous about that to me, since all the examples have an equal number of axes. We could add extra axes and broadcast the `order` array such that it fits the `data` array. I agree that it's a possible enhancement, but adding this would need more tests for higher-dimensional arrays. I think we can simplify our lives here and just throw an error saying that this isn't supported until someone actually needs exactly this?

Reply:
Regardless of what type the elements are, the array is just re-arranged according to the order on the first level.

So if for example you have `rearrange(data = [[2,3], [4,3]], order=[1,0])`, you'd get `[[4,3], [2,3]]`. Hope that helps.

Reply:
> Regardless of what type the elements are, the array is just re-arranged according to the order on the first level.
> 
> So if for example you have `rearrange(data = [[2,3], [4,3]], order=[1,0])`, you'd get `[[4,3], [2,3]]`. Hope that helps.

Thanks for this comment - so order is always expected to only have a single axis?

Reply:
Yes, indeed. It is of type `array<integer>` so can't be multi-dimensional.